python/ray/data/datasource/filename_provider.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Dict, Optional
+
+from ray.data.block import Block
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class FilenameProvider:
+    """Generates filenames when you write a :class:`~ray.data.Dataset`.
+
+    Use this class to customize what your written files look like.
```

Comment:
```suggestion
    Use this class to customize the filenames used when writing a Dataset.
```

(the previous wording made it sound like it was about the file content).